File: Java Old Code/Class8/src/perfectnumber.java
import java.io.*;
public class perfectnumber
{
    public static void main ( String args [])throws IOException
    { 
        DataInputStream dis = new DataInputStream(System.in);
        System.out.println("ENTER A NUMBER");
       int a = Integer.parseInt(dis.readLine());
       int b= 0 ;
       int i;
       int f = 0 ;
              for( i=1;i<=a-1;i++)
       
          
          if(a%i == 0)
          {
              b = b + i;
            }
        
         if (  b == a)
         {
         System.out.println( a + " is a perfect number");
        }
         else
         {
           System.out.println( a + " is not a perfect number");
        }
}


 }                                                                                                     
 
 
   

File: C_Programs/PerfectNumber.c
#include <stdio.h>
int main()
{
    int num;
    printf("Enter A Number:");
    scanf("%d",&num);
    int sum = 0;
    for (int i = 1; i < num; i++)
    {
        if (num%i==0)
            sum+=i;
    }
    if(sum==num)
        printf("%d is a Perfect Number",num);
    else
        printf("%d is not a Perfect Number",num);
    return 0;
}   